### PR TITLE
Pragmatical/1019 fluxinit kustomize

### PR DESCRIPTION
**Pull request by pragmatical**
# Type of PR

- [ ] Documentation changes
- [ ] Code changes
- [X] Test changes
- [X] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] My code follows the code style of this project.
- [X] I ran lint checks locally prior to submission.
- [X] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR
pulled gotk-components for flux into base directory and created kustomizations for preprod

## Does this introduce a breaking change

- [ ] YES
- [X] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #1019

### CODE REVIEW

**Review by gortegaMS — CHANGES_REQUESTED** [review]
base\gotk-components.yaml  uses the ACR name from DEV.  since this is the base file 
should it be PLACEHOLDER instead? if so then init-dev\gotk-sync.yaml should defined the patch similar to init-prod.
Just request changes to block it so we can discuss.

File: flux-init/init-preprod/kustomization.yaml [hunk 1]
```diff
@@ -1,6 +1,39 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
-- gotk-components-pre.yaml
+- ../base
 - gotk-repo.yaml
 - gotk-sync.yaml
+patches:
```
Comment (gortegaMS) [inline flux-init/init-preprod/kustomization.yaml:7]: Seems like controller's patch definition is the same as the one in init-prod\gotk-sync.yaml

**Review by gortegaMS — CHANGES_REQUESTED** [review]

File: flux-init/init-preprod/kustomization.yaml (hunk 1 above)
Comment (pragmatical) [reply flux-init/init-preprod/kustomization.yaml:7]: good catch... gonna leave just in this file

**Comment by kforeverisback** [comment]
@gortegaMS @AAkindele I think the base acr should be pointing to a docker/ghcr instance which is base without any ACR instance. Base flux images are in ghcr.io, see [this sample branch](https://github.com/retaildevcrews/ngsa-asb/commit/3899899647b4f4cea95ada53d75620c25b76497e) for example.

> base\gotk-components.yaml uses the ACR name from DEV. since this is the base file should it be PLACEHOLDER instead? if so then init-dev\gotk-sync.yaml should define the patch for it similar to init-prod. Just request changes to block it so we can discuss.

**Comment by kforeverisback** [comment]
**This is optional**
but should we put `gotk-sync.yaml` and `gotk-repo.yaml` into base dir?
Both have all the values common except for the `path:` in `gotk-sync.yaml` which can be patched.

Thoughts?

File: flux-init/init-preprod/kustomization.yaml (hunk 1 above)
Comment (pragmatical) [reply flux-init/init-preprod/kustomization.yaml:5]: Have addressed this issue

**Review by gortegaMS — APPROVED** [review]
LGTM

**Pull request merged**